nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/MapRecord.java

```diff
@@ -681,22 +752,18 @@ private RecordField getUpdatedRecordField(final RecordField field) {
             }
 
             final DataType mergedDataType = RecordFieldType.CHOICE.getChoiceDataType(updatedPossibleTypes);
-            return new RecordField(field.getFieldName(), mergedDataType, field.getDefaultValue(), field.getAliases(), field.isNullable());
+            return new RecordField(specField.getFieldName(), mergedDataType, specField.getDefaultValue(), specField.getAliases(), specField.isNullable());
         }
 
-        return field;
+        return specField;
     }
 
     private boolean isSimpleType(final RecordFieldType fieldType) {
-        switch (fieldType) {
-            case ARRAY:
-            case RECORD:
-            case MAP:
-            case CHOICE:
-                return false;
-        }
+        return switch (fieldType) {
+            case ARRAY, RECORD, MAP, CHOICE -> false;
+            default -> true;
+        };
```

Comment:
We absolutely should adopt language niceties as we go forward and hopefully backport considerations don't cause us to avoid that.  Backports are fair game where it makes sense but should happen with specific PRs and this will be increasingly true as we go forward.

Reply:
No, I have no plans to backport.